docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -234,316 +317,405 @@ async function run() {
 
-:::tip Get your results now or later
+Instead of using your actual Activities, you replace the Activities `getIP` and `getAddress`  with async functions that return hard-coded values. This way you're testing the Workflow's logic independently of the Activities. If you wanted to test the Activities directly as part of an integration test, you'd specify them directly as you did when you wrote the Worker program.
 
-You can [get the results](https://docs.temporal.io/dev-guide/typescript/foundations/#get-workflow-results) from your Workflow right away, or you can get the results at a later time. This implementation attempts to get the results immediately by logging the output of the `example` Workflow which in turn calls the `greet` Activity.
-:::
+Running the tests requires using the `mocha` command along with requiring the following libraries and pointing the test runner to the appropriate folder. Here's what the command would look like to run the tests:
 
-Now that your client code is written, it's time for you to use this code to start your Workflow Execution.
+```command
+npx  mocha \
+  --require ts-node/register \
+  --require source-map-support/register \
+  src/mocha/*.test.ts
+```
 
-## Start a Workflow Execution
+However, since this is quite verbose to type into the command line every time, you'll find a `test` script defined in `package.json` that runs this command for you.
 
-To start your Workflow, open a new tab in your terminal and run the `npm` script:
+Ensure you've saved all of your files and execute your tests with the following command:
 
 ```command
-npm run workflow
+npm test
 ```
 
-The script runs and prints the result:
+The test environment starts, spins up a Worker, and executes the Workflow in the test environment. At the end, you'll see that your test passes:
 
-```
-Hello, Temporal!
+```output
+    ✓ successfully completes the Workflow with mocked Activities (543ms)
+
+  1 passing (723ms)
 ```
 
-The "Hello, Temporal!" output is generated as the result of a few steps. First, `client.ts` passes `Temporal` as an argument to the Workflow. Next, the Workflow passes the argument to the Activity, and finally the Activity returns `Hello, ${name}!`.
+:::tip
+If you get an error that your test has timed out, run it again. The first time you run the tests on your local machine, the test server might not start quickly enough.
+:::
 
-You have successfully built a Temporal application from scratch!
+With a Workflow test in place, you can write unit tests for the Activities.
 
-## ![](/img/icons/check.png) Add a unit test
+Both of your Activities make external calls, and it'll be difficult to write tests that reliably test your Activities if they make calls to external services because you'll get a different IP address based on where your machine is located. To ensure the Activities are testable in isolation, you'll stub out the `fetch` calls using the sinon library.
```

Comment:
Do you want to add any description of sinon, even if it's just "sinon is a JavaScript testing library that provides mocks for situations like this one"?